feat(prisma): models for User and Session

Anfiu123 opened the pull request:
I made the models for User and Session and i also added Servers, Rules, Member, Channels, Category and serverRole

Anfiu123 committed:
feat(prisma): models for User and Session

I made the models for User and Session and i also added Servers, Rules, Member, Channels, Category and serverRole

r3b00thx commented on prisma/schema.prisma:
```diff
@@ -6,3 +6,132 @@ datasource db {
+model Member {
+  id         String @id @default(auto()) @map("_id") @db.ObjectId
+  uuid       String @unique @default(uuid())
+  serverUuid String
```
Where is the relation between the member and the server?

r3b00thx commented on prisma/schema.prisma:
```diff
@@ -6,3 +6,132 @@ datasource db {
+  serverUuid String
+
+  banned   Boolean @default(false)
+  ipBanned Json
```
What is this for? Please add documentation so other developers can understand what you are doing over here

r3b00thx commented on prisma/schema.prisma:
```diff
@@ -6,3 +6,132 @@ datasource db {
+  category    String
+  role        Json
+  type        String
+  permission  Json
```
`permission` -> `permissions`

r3b00thx commented on prisma/schema.prisma:
```diff
@@ -6,3 +6,132 @@ datasource db {
+
+  title      String
+  channels   Json
+  permission Json
```
Same here

r3b00thx commented on prisma/schema.prisma:
```diff
@@ -6,3 +6,132 @@ datasource db {
+  updatedAt DateTime @updatedAt
+}
+
+model serverRole {
```
I will send you to Jesus, where in the world a proper model starts with a lower case character?

r3b00thx commented on prisma/schema.prisma:
```diff
@@ -6,3 +6,132 @@ datasource db {
+  nickname     String?
+  bio          String?
+  image        String
+  servers      Json?
```
Where is the relation in here?

r3b00thx commented on prisma/schema.prisma:
```diff
@@ -6,3 +6,132 @@ datasource db {
+  bio          String?
+  image        String
+  servers      Json?
+  friends      Json?
```
Same here

r3b00thx commented on prisma/schema.prisma:
```diff
@@ -6,3 +6,132 @@ datasource db {
+model Session {
+  id      String @id @default(auto()) @map("_id") @db.ObjectId
+  uuid    String @unique @default(uuid())
+  ip      String
```
`ip` -> `ipAddress`

Anfiu123 committed:
fix(prisma): wrong naming conventions and refactor prisma to use relations

I changet all Jsons with relations, i changed all the wrong conventions, i rewrote the logi of the categories and members and i change the inviteLink type from Json to String.

itsvlxd commented on prisma/schema.prisma:
```diff
@@ -6,3 +6,135 @@ datasource db {
+
+  members  Member[]
+  roles    Role[]
+  category Category[]
+
+  createAt  DateTime @default(now())
+  updatedAt DateTime @updatedAt
+}
```
Conventionally, it's createdAt, not createAt. Same for updatedAt.

itsvlxd commented on prisma/schema.prisma:
```diff
@@ -6,3 +6,135 @@ datasource db {
+
+  session Session[]
+  member  Member[]
+
+  createAt  DateTime @default(now())
+  updatedAt DateTime @updatedAt
+}
+
+model Session {
+  id      String @id @default(auto()) @map("_id") @db.ObjectId
+  uuid    String @unique @default(uuid())
+  ipAddress      String
+  browser String
+
+  sessionToken String   @unique
+  userUUID     String
+  expires      DateTime
+
+  user User @relation(fields: [userUUID], references: [uuid])
+
+  createAt  DateTime @default(now())
+  updatedAt DateTime @updatedAt
+}
```
Conventionally, it's createdAt, not createAt. Same for updatedAt.

itsvlxd commented on prisma/schema.prisma:
```diff
@@ -6,3 +6,135 @@ datasource db {
+  inviteLink String
+
+  ownerUUID String
+
+  members  Member[]
+  roles    Role[]
+  category Category[]
+
```
members.....yay
roles.............yay
category......really? 0-0

itsvlxd commented on prisma/schema.prisma:
```diff
@@ -6,3 +6,135 @@ datasource db {
+  uuid String @unique @default(uuid())
+
+  serverUUID     String
+  permissionUUID String
+
+  name       String
+  permission CategoryPermission @relation(fields: [permissionUUID], references: [uuid])
+
+  server   Server    @relation(fields: [serverUUID], references: [uuid])
+  channels Channel[]
+}
+
+model Channel {
+  id   String @id @default(auto()) @map("_id") @db.ObjectId
+  uuid String @unique @default(uuid())
+
+  categoryUUID   String
+  permissionUUID String
+
+  name       String
+  permission CategoryPermission @relation(fields: [permissionUUID], references: [uuid])
+  type       String
+
+  category Category @relation(fields: [categoryUUID], references: [uuid])
+}
+
+model CategoryPermission {
+  id   String @id @default(auto()) @map("_id") @db.ObjectId
+  uuid String @unique @default(uuid())
+
+  name      String
+  canView   Boolean @default(true)
+  canSend   Boolean @default(true)
+  canManage Boolean @default(false)
+  canDelete Boolean @default(false)
+
+  category Category[]
+  channel  Channel[]
+}
+
```
Consider a base Permission model and extend it logically.

itsvlxd commented on prisma/schema.prisma:
```diff
@@ -6,3 +6,135 @@ datasource db {
+  categoryUUID   String
+  permissionUUID String
+
+  name       String
+  permission CategoryPermission @relation(fields: [permissionUUID], references: [uuid])
+  type       String
+
+  category Category @relation(fields: [categoryUUID], references: [uuid])
+}
```
Enum Suggestion for Channel.type and User.status

r3b00thx commented:
@Anfiu123 commit?
